pkg/cloud/converters/tags.go

```diff
@@ -49,14 +51,21 @@ func MapPtrToMap(src map[string]*string) infrav1.Tags {
 	return tags
 }
 
-// MapToTags converts a infrav1.Tags to a []*ec2.Tag.
+// MapToTags converts infrav1.Tags to []*ec2.Tag.
 func MapToTags(src infrav1.Tags) []*ec2.Tag {
 	tags := make([]*ec2.Tag, 0, len(src))
 
-	for k, v := range src {
+	// For testing, we need sorted keys
+	sortedKeys := make([]string, 0, len(tags))
```

Comment:
We already have logic to sort tags in other unit tests. Refer [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/d9bb1cfd2c7fc94a437efafbc89486dfd22d361f/pkg/cloud/services/ec2/helper_test.go#L186-L190) for the implementation and you can check how it's used in [launchtemplate_test.go](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/d9bb1cfd2c7fc94a437efafbc89486dfd22d361f/pkg/cloud/services/ec2/launchtemplate_test.go#L676)
IMO we should not modify code for testing purposes.